### PR TITLE
chore: Upgrade go.mod dependencies, fix jose v2 vulnerability, upgrade jemalloc

### PR DESCRIPTION
**Pull request by RJKeevil**
Description: A general upgrade to latest (compatible) versions of multiple dependencies. Jose v2 also triggers vulnerability issues in our containers, so replaced that with v4 everywhere.

### CODE REVIEW

**Comment by mangalaman93** [comment]
Thank you for the PR. Seems like one test is failing. I am gonna look into it.

File: go.mod [hunk 1]
```diff
@@ -42,30 +43,29 @@ require (
 	github.com/paulmach/go.geojson v1.5.0
 	github.com/pkg/errors v0.9.1
 	github.com/pkg/profile v1.7.0
-	github.com/prometheus/client_golang v1.20.2
+	github.com/prometheus/client_golang v1.20.3
 	github.com/soheilhy/cmux v0.1.5
-	github.com/spf13/cast v1.3.1
+	github.com/spf13/cast v1.7.0
```
Comment (mangalaman93) [inline go.mod:48]: please do not update these deps. There is a bug in cast library causing our tests to fail. More details here https://github.com/dgraph-io/dgraph/pull/9141

**Review by mangalaman93 — CHANGES_REQUESTED** [review]

**Comment by RJKeevil** [comment]
@mangalaman93 chnages made, hopefully tests pass now, lets see

**Review by mangalaman93 — APPROVED** [review]

**Comment by RJKeevil** [comment]
@mangalaman93 missing TLS cert, do you think this is also caused by an incompatible module upgrade?

**Pull request merged**